### ISSUE TITLE
Algorithms can be notebooks, predictive matrices?

### ISSUE THREAD

**Issue by markpyzhov** [MallCloud/contracts-api#12]
The original text of requirements is described in [wiki/BlockChain-communication](https://github.com/MallCloud/contracts-api/wiki/BlockChain-communication#business-requirements).

> 1. ... Algorithms can be notebooks, predictive matrices, etc.

I don't understand how the algorithm can be the predictive matrices. Can you explain it please?

**Comment by ehillerbrand** [comment]
I should have said notebooks are predictive matrices.  My mistake.

But as a DS I could load the algorithm or load the predictive matrix.  Agree?

**Comment by markpyzhov** [comment]
> But as a DS I could load the algorithm or load the predictive matrix. Agree?

Can you explain use case when DS uploads predictive matrix? 

I thought DS can upload DataSet to improve training and prediction process, by expanding the canonical schema or expanding amount of rows in data set.

**Comment by ehillerbrand** [comment]
You are correct:  I thought DS can upload DataSet to improve training and prediction process, by expanding the canonical schema or expanding amount of rows in data set.

**Comment by ehillerbrand** [comment]
Here is the use case:
1.  A DS has a dataset on their personal server (lets say it is a weather database).
2.  The DS executes an algorithm against the data set (the algorithm predicts snow)
3.  The DS optimizes the algorithm and the result of the algorithm is a prediction matrix (NOT PREDICTIONS).  It is the result of a training process.  For HMM the prediction matrix is called a _transition matrix_ (https://www.quora.com/What-is-Hidden-Markov-Model-What-is-the-output-of-it-and-How-to-calculate-the-state-transition-probability)
4.  The DS loads the transition matrix and the notebook that executes the transition matrix.  IN order to do this, the DS would need to map the algorithm inputs and outputs to the graph database.

In other words, the DS is loading an already created and trained model that is the result of the training process.  In the case of the HMM, the _transition matrix_ is a data matrix.
